draft/spec/index.html

```diff
@@ -340,6 +340,47 @@ <h2>Object Structure</h2>
         ├── inventory.json
         └── inventory.json.sha512
         </pre>
+
+        <section id="object-interoperability">
+            <h2>Interoperability</h2>
+            <p>
+                Within an OCFL Object and its associated <a>Inventory</a>, the following restrictions are applied
+                to filesystem features. These restrictions are provided in an attempt to normalize file handling
+                behaviors and addressability of files within the Inventory.
+            </p>
+            <p>
+                If use of these features are required to preserve an exact copy of the content (for example,
+                preserving a hard-drive image), implementers SHOULD consider using a packaging format,
+                such as ZIP or TAR files or ISO disk images, to preserve the content. See the
+                [[OCFL-Implementation-Notes]] for further details.
+            </p>
+            <ul>
+                <li>
+                    <strong>Symbolic links</strong> (aka 'shortcuts' or 'aliases') MUST NOT be used within an
+                    OCFL Object. These are not portable and may lead to undefined behaviours in client applications
+                </li>
+                <li>References to <strong>file paths</strong> within the <a>inventory</a> MUST use the "forward-slash"
+                    ('/') as a path separator. Some filesystem use other characters such as the "back-slash" ('\')
+                    or "colon" (':') as a path delimiter.
+                </li>
+                <li>
+                    <strong>File permissions</strong> MAY be applied to files in an OCFL Object; however, implementers
+                    should be aware that these are not portable across filesystems, and so should not be relied on
+                    to protect sensitive content. This is also the case for extended filesystem flags, such as
+                    Access Control Lists or Hidden files.
+                </li>
+                <li>
+                    The <strong>character encoding</strong> of the filesystem and Inventory SHOULD be
+                    Unicode-compatible, either UTF-8, UTF-16, or UCS-2. Implementers may experience problems
```

Comment:
JSON itself is defined over UTF-8, UTF-16 or UTF-32 (with either by order for UTF-16 or UTF-32), see https://tools.ietf.org/html/rfc4627#section-3 . I do not know whether all JSON parsers have good support for all of these encodings. There is no note about UCS-2 encoding so I'm not sure how we imagine that to be handled? I feel a little out of my comfort zone with this question but I think we need to be more explicit and tie to the JSON spec. The current  text seems to raise more questions than it answers in my mind.

Reply:
Hi, I'm new here but wanted to add that it may be helpful to be more specific about Unicode Normalization, particularly given the difficulties encountered by folks working on BagIt. (See [here](https://github.com/LibraryOfCongress/bagit-python/pull/82)).

Reply:
@rosy1280 I think the edge case to consider is when a repository is rebuilt on a file system that handles filenames differently than the filesystem the OCFL Object was created on. In that situation (as in the example with BagIt) it might be possible for the filenames in the inventory to differ from the actual filenames (even though they both look the same, visually).